### PR TITLE
use `posix_spawn`

### PR DESCRIPTION
**Pull request by tannernelson**

### CODE REVIEW

File: Sources/Console/Terminal/Terminal.swift [hunk 1]
```diff
@@ -58,60 +72,61 @@ public class Terminal: ConsoleProtocol {
...
     }
 
-    public func execute(_ command: String) throws {
-        #if os(Linux)
-            let input = FileHandle.standardInput()
-            let output = FileHandle.standardOutput()
-            let error = FileHandle.standardError()
+    public func execute(program: String, arguments: [String], input: Int32? = nil, output: Int32? = nil, error: Int32? = nil) throws {
+        var pid = UnsafeMutablePointer<pid_t>.allocate(capacity: 1)
+        pid.initialize(to: pid_t())
```
Comment (LoganWright) [inline Sources/Console/Terminal/Terminal.swift]: Do we need to release this, or is that happening elsewhere?

File: Sources/Console/Terminal/Terminal.swift [hunk 2]
```diff
@@ -58,60 +72,61 @@ public class Terminal: ConsoleProtocol {
...
-        let error = Pipe()
 
-        do {
-            try execute(command, input: input, output: output, error: error)
-        } catch ConsoleError.execute(let result) {
-            let error = String(data: error.fileHandleForReading.readDataToEndOfFile(), encoding: .utf8) ?? "Unknown"
-            throw ConsoleError.subexecute(result, error)
+        func getenv(_ key: String) -> String? {
+            let out = libc.getenv(key)
+            return out == nil ? nil : String(validatingUTF8: out!)  //FIXME locale may not be UTF8
```
Comment (LoganWright) [inline Sources/Console/Terminal/Terminal.swift]: You could also do `out.flatMap { String(validatingUTF8: $0) }` and remove the `!`

File: Sources/Console/Terminal/Terminal.swift (hunk 1 above)
Comment (tannernelson) [inline Sources/Console/Terminal/Terminal.swift]: Oops, yes we do.

Comment (tannernelson) [inline Sources/Console/Terminal/Terminal.swift]: It's now being deallocated in the line afterward.

**Pull request merged**